Fix Bang & Olufsen event testing

mj23000 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

Currently the `test_button_event_creation_beoconnect_core` uses the wrong event entity_id prefix, which caused all of the entity_ids to be malformed compared to what was expected. The test checks if entities haven't been created, so it didn't fail.

 `beosound_balance_11111111_` was used, `beosound_balance_22222222_` was expected and `beoconnect_core_22222222_` was intended.

Additionally `test_button_event_creation` didn't actually check if the entities were created.


This PR fixes the aforementioned bugs, uses _2 constants where they were expected and adds a check for the number of entities and a syrupy snapshot to ensure that this doesn't happen again.


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

mj23000 committed:
Fix core event entity_ids

Use _2 constants where applicable

mj23000 committed:
Add number of entities created and snapshot check

abmantis commented on tests/components/bang_olufsen/test_event.py:
```diff
@@ -44,12 +46,21 @@ async def test_button_event_creation(
+    entity_ids_available = list(entity_registry.entities.keys())
+    assert len(entity_ids_available) == 1 + len(entity_ids)
+
+    # Check snapshot
+    assert entity_ids_available == snapshot()
```
```suggestion
    assert entity_ids_available == snapshot
```

abmantis commented on tests/components/bang_olufsen/test_event.py:
```diff
@@ -69,6 +80,14 @@ async def test_button_event_creation_beoconnect_core(
+    entity_ids_available = list(entity_registry.entities.keys())
+    assert len(entity_ids_available) == 1
+
+    # Check snapshot
+    assert entity_ids_available == snapshot()
```
```suggestion
    assert entity_ids_available == snapshot
```

abmantis commented on tests/components/bang_olufsen/test_event.py:
```diff
@@ -69,6 +80,14 @@ async def test_button_event_creation_beoconnect_core(
     for entity_id in entity_ids:
         assert not entity_registry.async_get(entity_id)
 
```
since we already check that only 1 entity exists, can we remove this loop?

mj23000 replied:
Done

abmantis commented on tests/components/bang_olufsen/test_event.py:
```diff
@@ -44,12 +46,21 @@ async def test_button_event_creation(
     for entity_id in entity_ids:
         entity_registry.async_get(entity_id)
```
just noticed that there is no `assert` or check here. is this actually testing anything? (I don't think the method throws, or it would be throwing in L81)

mj23000 replied:
Yup. Looks like this test didn't work either.

mj23000 committed:
Remove () from snapshot

Co-authored-by: Abílio Costa <abmantis@users.noreply.github.com>

mj23000 committed:
Fix test not properly checking entity availability

Fix entity_id using preset_ instead of favourite_
Remove unnecessary entity check

mj23000 committed:
Remove unnecessary entity enable step

abmantis reviewed (approved):
Thanks!
